datamodel_code_generator/parser/jsonschema.py

```diff
@@ -550,41 +553,64 @@ def parse_enum(
 
     def parse_ref(self, obj: JsonSchemaObject, path: List[str]) -> None:
         if obj.ref:
-            ref: str = obj.ref
-            # https://swagger.io/docs/specification/using-ref/
-            if obj.ref.startswith('#'):
-                # Local Reference – $ref: '#/definitions/myElement'
-                pass
-            elif '://' in ref:
-                # URL Reference – $ref: 'http://path/to/your/resource' Uses the whole document located on the different server.
-                raise NotImplementedError(f'URL Reference is not supported. $ref:{ref}')
-
-            else:
-                # Remote Reference – $ref: 'document.json' Uses the whole document located on the same server and in
-                # the same location. TODO treat edge case
-                relative_path, object_path = ref.split('#/')
-                full_path = self.base_path / relative_path
-                with full_path.open() as f:
-                    if full_path.suffix.lower() == '.json':
-                        import json
-
-                        ref_body: Dict[str, Any] = json.load(f)
+            reference = self.model_resolver.get(obj.ref)
+            if not reference or not reference.loaded:
+                # https://swagger.io/docs/specification/using-ref/
+                if obj.ref.startswith('#'):
+                    # Local Reference – $ref: '#/definitions/myElement'
+                    pass
+                else:
+                    relative_path, object_path = obj.ref.split('#/')
+                    remote_object: Optional[
+                        Dict[str, Any]
+                    ] = self.remote_object_cache.get(relative_path)
+                    if obj.ref.startswith(('https://', 'http://')):
+                        full_path: Union[Path, str] = relative_path
+                        if remote_object:
+                            ref_body: Dict[str, Any] = remote_object
+                        else:
+                            # URL Reference – $ref: 'http://path/to/your/resource' Uses the whole document located on the different server.
+                            try:
+                                import httpx
+                            except ImportError:  # pragma: no cover
+                                raise Exception(
+                                    f'Please run $pip install datamodel-code-generator[http] to resolve URL Reference ref={obj.ref}'
+                                )
+                            raw_body: str = httpx.get(relative_path).text
+                            if relative_path.lower().endswith('.json'):
+                                import json
+
+                                ref_body = json.loads(raw_body)
+                            else:
+                                # expect yaml
+                                import yaml
+
+                                ref_body = yaml.safe_load(raw_body)
```

Comment:
```suggestion
                            import yaml
                            ref_body = yaml.safe_load(raw_body)
```
Why don't always import yaml? Yaml parser can parse json too.

Reply:
I don't know the feature in YAML parser. 
Thank you.